primer/brand
# SubNav v2

## Summary

Part of https://github.com/github/primer/issues/3312

Updates `SubNav` component to support anchor-based navigation similar to `AnchorNav`.

⚠️ Despite being internally referred to as `SubNav v2` there are no breaking changes to the API as part of this update. A full visual refresh however, has been implemented per latest Site guidance.

🔗  [Documentation](https://primer-dbe99eb951-26139705.drafts.github.io/brand/components/SubNav#optional-anchor-based-sub-menu)
🔗  [Storybook examples](https://primer-dbe99eb951-26139705.drafts.github.io/brand/storybook/?path=/story/components-subnav-features--dropdown-variant)



## List of notable changes:



- updated `SubNav.Submenu` to support `variant="anchor"` 
- updated Storybook examples to respect color modes and differentiate between dropdown and anchor submenu types
- fixed various bugs in existing subnav like not closing mobile overlays when clicking outside it

## What should reviewers focus on?



- Manually review each of the Storybook examples for design accuracy
- Read through the docs and verify new features are accurately captured

## Steps to test:



1. Use [these Storybook examples](https://primer-dbe99eb951-26139705.drafts.github.io/brand/storybook/?path=/story/components-subnav-features--dropdown-variant)
1. Use [this docs link](https://primer-dbe99eb951-26139705.drafts.github.io/brand/components/SubNav#optional-anchor-based-sub-menu)
1. Use the visual diffs to see before and afters

## Supporting resources (related issues, external links, etc):

- [Tracking issue](https://github.com/github/primer/issues/3312)

## Contributor checklist:


## Reviewer checklist:


## Screenshots:

> Please try to provide before and after screenshots or videos

<table>
<tr>
<th> Before</th> <th> After </th>
</tr>
<tr>
<td valign="top">

![image](https://github.com/user-attachments/assets/9f15cbdf-f6cd-4cfd-b8f4-0e3301579a33)

 </td>
<td valign="top">

![image](https://github.com/user-attachments/assets/55b4092b-3561-4cf0-8999-6d76c8a70e1e)

</td>
</tr>
</table>

New anchor navigation

<table>
<tr>
<th> Light</th> <th> Dark </th>
</tr>
<tr>
<td valign="top">


![Screenshot 2024-11-26 at 10 49 38](https://github.com/user-attachments/assets/7e62339a-b873-46d1-9584-16592fa7cb83)

 </td>
<td valign="top">


![Screenshot 2024-11-26 at 10 49 44](https://github.com/user-attachments/assets/91b5c37a-7b16-4425-960b-bd2dcf4fad0c)

</td>
</tr>
</table>
